src/Cms/PageRules.php

```diff
@@ -293,6 +297,25 @@ public static function update(Page $page, array $content = []): bool
             ]);
         }
 
+        self::slugMaxlength($page);
+
         return true;
     }
+
+    protected static function slugMaxlength(Page $page, string $slug = null): void
```

Comment:
## 1. Naming

I think the name of the method should be `validateSlugLength` to make it clearer what it does.

## 2. Method signature

The second argument should be explicitly nullable:

```suggestion
    protected static function validateSlugLength(Page $page, ?string $slug = null): void
```

As the method doesn't always need the page (only if the slug is not given), the cleanest way would be to have only one argument, but type hinting both a page and a string is not possible until PHP 8.

Alternative: Only ever accept a string with the slug and make the other methods pass that explicitly:

```suggestion
    protected static function validateSlugLength(string $slug): void
```

I prefer the second alternative, but that's just personal preference.

## 3. Documentation

Could you also add a method docblock above it? All methods will have one at some point, so it's best to add them to new methods already while we still now what the method was supposed to do.

```php
/**
 * Ensures that the slug doesn't exceed the maximum length to make
 * sure that the directory name will be accepted by the filesystem
 *
 * @param string $slug New slug to check
 * @return void
 *
 * @throws \Kirby\Exception\InvalidArgumentException If the slug is too long 
 */
```